receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_others.go

```diff
@@ -66,11 +88,17 @@ func (s *scraper) ScrapeMetrics(_ context.Context) (pdata.MetricSlice, error) {
 		return metrics, err
 	}
 
-	metrics.Resize(3 + systemSpecificMetricsLen)
-	initializeDiskIOMetric(metrics.At(0), s.startTime, ioCounters)
-	initializeDiskOpsMetric(metrics.At(1), s.startTime, ioCounters)
-	initializeDiskTimeMetric(metrics.At(2), s.startTime, ioCounters)
-	appendSystemSpecificMetrics(metrics, 3, s.startTime, ioCounters)
+	// filter devices by name
+	ioCounters = filterByDevice(ioCounters)
```

Comment:
Not sure how this compile, is this an error?

Reply:
Yes